deepmd/tf/nvnmd/entrypoints/mapt.py

```
@@ -566,18 +604,22 @@ def build_t2g(self):
         ebd_type = self.build_embedding_net(dic_ph["t_one_hot"], wbs, None)
-        last_type = tf.cast(tf.zeros([1, ebd_type.shape[1]]), filter_precision)  # pylint: disable=no-explicit-dtype
+        last_type = tf.cast(tf.zeros([1, ebd_type.shape[1]]), filter_precision)
         ebd_type = tf.concat([ebd_type, last_type], 0)
         dic_ph["t_ebd"] = ebd_type
         # type_embedding of i, j atoms -> two_side_type_embedding
         type_embedding = dic_ph["t_ebd"]
         padding_ntypes = type_embedding.shape[0]
         type_embedding_nei = tf.tile(
-            tf.reshape(type_embedding, [1, padding_ntypes, -1]),
+            tf.reshape(
+                type_embedding, [1, padding_ntypes, -1], dtype=GLOBAL_TF_FLOAT_PRECISION
```

Ruff (0.11.9)

Undefined name `GLOBAL_TF_FLOAT_PRECISION`

(F821)